### PR TITLE
[WIP] WINC-827: Resolve Windows service PowerShell variables

### PR DESCRIPTION
**Pull request by sebsoto**
Resolves PowerShell variables present in the Windows service configmap
when starting Windows services.

### CODE REVIEW

File: pkg/daemon/controller/controller.go [hunk 1]
```diff
@@ -370,11 +393,18 @@ func (sc *ServiceController) resolveNodeVariables(svc servicescm.Service) (map[s
...
-// resolvePowershellVariables returns a map, with the keys being each variable, and the value being the string to replace the
-// variable with
+// resolvePowershellVariables returns a map, with the keys being each variable, and the value being the string to
+// replace the variable with
 func (sc *ServiceController) resolvePowershellVariables(svc servicescm.Service) (map[string]string, error) {
-	// TODO: Implement this function
-	return make(map[string]string), nil
+	vars := make(map[string]string)
+	for _, psVar := range svc.PowershellVariablesInCommand {
+		out, err := sc.psCmdRunner.Run(psVar.Path)
```
Comment (jrvaldes) [inline pkg/daemon/controller/controller.go:401]: For executing PowerShell scripts from a file `powershell.exe -File` may be better equipped. See PowerShell [docs](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_powershell_exe?view=powershell-5.1#-file----filepath-args).

With that, I feel you may need two APIs for `psCmdRunner`:
- psCmdRunner.Run
- psCmdRunner.RunFile

**Review by jrvaldes — COMMENTED** [review]
@sebsoto  Thanks for working on this, comments added. PTAL

File: pkg/daemon/controller/controller.go [hunk 2]
```diff
@@ -138,9 +144,26 @@ func RunController(ctx context.Context, apiServerURL, saCA, saToken string) erro
...
+	}
+	return &ServiceController{client: client, Mgr: mgr, ctx: ctx, psCmdRunner: cmdRunner,
+		nodeName: nodeName, watchNamespace: wmcoNamespace}
+}
+
+// psCmdRunner implements the PowershellCommandRunner interface
+type psCmdRunner struct{}
+
+// Run runs the command with the PowerShell on PATH
+func (r *psCmdRunner) Run(cmd string) (string, error) {
```
Comment (saifshaikh48) [inline pkg/daemon/controller/controller.go:161]: nit: I'd prefer the `psCmdRunner` struct and its method(s) be moved closer to the interface it implements for readability reasons

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm aside from one suggestion

**Comment by sebsoto** [comment]
After a discussion with the team this code will be merged as part of moving hybrid-overlay into WICD, as this will allow for e2e testing.

**Pull request closed without merging**